app/controllers/expeditions_controller.rb

```diff
@@ -23,12 +24,22 @@ def create
 
   def destroy
     @expedition = Expedition.find(params[:id])
+    claim_prize
     @expedition.destroy
     redirect_to expeditions_index_path(current_user.id)
   end
 
   private
 
+  def claim_prize
+    prize_resources = ExpeditionPrize.where(expedition_type_id: @expedition.expedition_type)
+    prize_resources.each do |prize_resource|
+      @resource = current_user.resources.find_by(resource_type: prize_resource.resource_type.id)
```

Comment:
Nie zawsze user będzie miał to resource. `current_user.resources.find_by(resource_type: prize_resource.resource_type.id)` może zwrócić `nil`